rust/matches/src/lib.rs

```diff
@@ -999,6 +999,12 @@ pub struct Root {
     root: [u8; 32],
 }
 
+impl Root {
+    pub fn space() -> usize {
```

Comment:
Why did you go with a function vs a const?

Reply:
Ive seen it done this way before, I kinda like it but if you dont I can change it. No one has done any impl for space so ive been going with this for my previous PRs

Reply:
I also like having an impl for space, but why is it a function? I've been doing them as simple consts. https://github.com/raindrop-studios/rain-redemptions/blob/0ea1720fa9ec979516e85eefbb969024aa57e4b0/programs/redemptionV0/src/lib.rs#L1250 . Functions incur additional overhead, and you don't have any functional requirements for the space calculation.